dali/kernels/signal/resampling_test.cc

```diff
@@ -16,93 +16,149 @@
 #include <vector>
 #include <numeric>
 #include "dali/kernels/signal/resampling.h"
+#include "dali/kernels/signal/resampling_test.h"
 
 namespace dali {
 namespace kernels {
 namespace signal {
 namespace resampling {
-
-namespace {
+namespace test {
 
 double HannWindow(int i, int n) {
   assert(n > 0);
-  return Hann(2.0*i / n - 1);
+  return Hann(2.0 * i / n - 1);
 }
 
-template <typename T>
-void TestWave(T *out, int n, int stride, float freq) {
-  for (int i = 0; i < n; i++) {
-    float x = i * freq;
-    float f = std::sin(i* freq) * HannWindow(i, n);
-    out[i*stride] = ConvertSatNorm<T>(f);
+void ResamplingTest::PrepareData(int nsamples, int nchannels, span<const Args> args, int nsec) {
+  TensorListShape<> in_sh(nsamples, nchannels > 1 ? 2 : 1);
+  TensorListShape<> out_sh(nsamples, nchannels > 1 ? 2 : 1);
+  for (int s = 0; s < nsamples; s++) {
+    double in_rate = args[s].in_rate;
+    double out_rate = args[s].out_rate;
+    double scale = in_rate / out_rate;
+    int n_in = nsec * in_rate + 12345 * s;  // different lengths
+    int n_out = std::ceil(n_in / scale);
+    int64_t out_begin = args[s].out_begin > 0 ? args[s].out_begin : 0;
+    int64_t out_end = args[s].out_end > 0 ? args[s].out_end : n_out;
+    ASSERT_GT(out_end, out_begin);
+    in_sh.tensor_shape_span(s)[0] = n_in;
+    out_sh.tensor_shape_span(s)[0] = out_end - out_begin;
+    if (nchannels > 1) {
+      in_sh.tensor_shape_span(s)[1] = nchannels;
+      out_sh.tensor_shape_span(s)[1] = nchannels;
+    }
+  }
+  ttl_in_.reshape(in_sh);
+  ttl_out_.reshape(out_sh);
+  ttl_outref_.reshape(out_sh);
+  for (int s = 0; s < nsamples; s++) {
+    double in_rate = args[s].in_rate;
+    double out_rate = args[s].out_rate;
+    double scale = static_cast<double>(in_rate) / out_rate;
```

Comment:
```suggestion
    double scale = in_rate / out_rate;
```
in_rate is already `double`.

Reply:
Fixed